server/src/main/kotlin/suwayomi/tachidesk/manga/impl/chapter/ChapterForDownload.kt

```diff
@@ -0,0 +1,148 @@
+        val mangaEntry = transaction { MangaTable.select { MangaTable.id eq mangaId }.first() }
+        val source = getCatalogueSourceOrStub(mangaEntry[MangaTable.sourceReference])
+
+        return source.fetchPageList(
+            SChapter.create().apply {
+                url = chapterEntry[ChapterTable.url]
+                name = chapterEntry[ChapterTable.name]
+            }
+        ).awaitSingle()
+    }
+
+    private fun markAsNotDownloaded() {
+        // chapter may be downloaded but if we are here, then images might be deleted and database data be false
+        transaction {
+            ChapterTable.update({ (ChapterTable.sourceOrder eq chapterIndex) and (ChapterTable.manga eq mangaId) }) {
+                it[isDownloaded] = false
+            }
+        }
+    }
+
+    private fun updateDatabasePages(pageList: List<Page>) {
+        val chapterId = chapterEntry[ChapterTable.id].value
+        val chapterIndex = chapterEntry[ChapterTable.sourceOrder]
+        val mangaId = chapterEntry[ChapterTable.manga].value
+
+        transaction {
+            pageList.forEach { page ->
+                val pageEntry = transaction {
+                    PageTable.select { (PageTable.chapter eq chapterId) and (PageTable.index eq page.index) }
+                        .firstOrNull()
+                }
+                if (pageEntry == null) {
+                    PageTable.insert {
+                        it[index] = page.index
+                        it[url] = page.url
+                        it[imageUrl] = page.imageUrl
+                        it[chapter] = chapterId
+                    }
+                } else {
+                    PageTable.update({ (PageTable.chapter eq chapterId) and (PageTable.index eq page.index) }) {
+                        it[url] = page.url
+                        it[imageUrl] = page.imageUrl
+                    }
+                }
+            }
+        }
+
+        updatePageCount(pageList, mangaId, chapterIndex)
+
+        // chapter was updated
+        chapterEntry = freshChapterEntry()
+    }
+
+    private fun updatePageCount(
+        pageList: List<Page>,
+        mangaId: Int,
+        chapterIndex: Int
+    ) {
+        val pageCount = pageList.count()
+
+        transaction {
+            ChapterTable.update({ (ChapterTable.manga eq mangaId) and (ChapterTable.sourceOrder eq chapterIndex) }) {
+                it[ChapterTable.pageCount] = pageCount
+            }
+        }
+    }
+
+    private fun isNotCompletelyDownloaded(): Boolean {
+        return !(chapterEntry[ChapterTable.isDownloaded] && firstPageExists())
+    }
+
+    private fun firstPageExists(): Boolean {
+        val chapterId = chapterEntry[ChapterTable.id].value
+
+        val chapterDir = getChapterDir(mangaId, chapterId)
+
+        println(chapterDir)
+        println(getPageName(0))
+
+        return ImageResponse.findFileNameStartingWith(
+            chapterDir,
+            getPageName(0)
```

Comment:
```suggestion
            getPageName(1)
```

Reply:
that's was a bug

It was checking for the second page